src/main/java/org/opentripplanner/routing/algorithm/raptor/transit/Transfer.java

```diff
@@ -36,29 +45,87 @@ public Transfer(
 
     public int getToStop() { return toStop; }
 
-    /**
-     * The effective distance is defined as the value that divided by the speed will give the
-     * correct duration of the transfer. This takes into account slowdowns/speedups related to
-     * slopes. It can also account for other factors that affect the transfer duration, but all
-     * factors are required to be proportional to the speed. The reason we are doing this is so
-     * that we can calculate all transfer durations in a reasonable amount of time for each
-     * incoming request.
-     */
-    public int getEffectiveWalkDistanceMeters() {
-        return effectiveWalkDistanceMeters;
+    public int getDistanceMeters() {
+        return distanceMeters;
     }
 
-    public int getDistanceIndependentTime() {
-        return distanceIndependentTime;
+    public List<Edge> getEdges() {
+        return edges;
     }
 
-    public int getDistanceMeters() {
-        return edges != null
-            ? (int) edges.stream().mapToDouble(Edge::getDistanceMeters).sum()
-            : effectiveWalkDistanceMeters;
+    public Optional<RaptorTransfer> asRaptorTransfer(RoutingRequest routingRequest) {
+        if (edges == null || edges.isEmpty()) {
+            int durationSeconds = (int) Math.ceil(distanceMeters / routingRequest.walkSpeed);
+            return Optional.of(new TransferWithDuration(
+                    this,
+                    durationSeconds,
+                    RaptorCostConverter.toRaptorCost((int) Math.ceil(durationSeconds * routingRequest.walkReluctance))
+            ));
+        }
+
+        StateEditor se = new StateEditor(routingRequest, edges.get(0).getFromVertex());
+        se.setTimeSeconds(0);
+
+        State s = se.makeState();
+        for (Edge e : edges) {
+            s = e.traverse(s);
+            if (s == null) {
+                return Optional.empty();
+            }
+        }
```

Comment:
In witch cases can this happen (doc) and why do we keep transfers that will never be used in routing?

Reply:
This happens when the given edge can't be traversed _with the current routing request_. An example is when the edges are not wheelchair accessible.